spec/hermes/consumer_builder_spec.rb

```diff
@@ -249,6 +252,67 @@ def call(event)
 
             process
           end
+
+          context "when error is raised during processing" do
+            context "when the error is related to Postgres server closing the connection" do
+              context "when the database connection provider is configured and distributed traces are supposed to be stored" do
+                let(:error_message) { "PG::ConnectionBad: PQsocket() can't get socket descriptor" }
+
+                before do
+                  allow(configuration.database_connection_provider.connection_pool).to receive(:disconnect!).and_call_original
+                  allow(Hermes::DistributedTrace.connection_pool).to receive(:disconnect!).and_call_original
+
+                  allow(Hermes::DependenciesContainer["event_processor"]).to receive(:call)
+                    .and_raise(ActiveRecord::StatementInvalid.new(error_message))
+                end
+
+                it { is_expected_block.to raise_error(ActiveRecord::StatementInvalid, error_message) }
+
+                it "releases DB connections" do
+                  process rescue ActiveRecord::StatementInvalid
+
+                  expect(configuration.database_connection_provider.connection_pool).to have_received(:disconnect!).at_least(1)
+                  expect(Hermes::DistributedTrace.connection_pool).to have_received(:disconnect!).at_least(1)
```

Comment:
why at least once?

Reply:
@adamgrad looks like this is called internally in ActiveRecord.